src/main/java/uk/ac/sanger/sccp/stan/service/register/SectionRegisterValidation.java

```diff
@@ -443,6 +447,40 @@ public UCMap<SlotRegion> validateRegions() {
         return slotRegions;
     }
 
+    public UCMap<BioRisk> validateBioRisks() {
+        boolean anyMissing = false;
+        Set<String> codes = new HashSet<>();
+        for (SectionRegisterLabware rl : request.getLabware()) {
+            for (SectionRegisterContent content : rl.getContents()) {
+                String code = content.getBioRiskCode();
+                if (code != null) {
+                    code = emptyToNull(code.trim());
+                    content.setBioRiskCode(code);
+                }
+                if (code == null) {
+                    anyMissing = true;
+                } else {
+                    codes.add(code);
+                }
+            }
+        }
+        if (anyMissing) {
+            addProblem("Missing bio risk.");
+        }
+        if (codes.isEmpty()) {
+            return new UCMap<>(0);
+        }
+        UCMap<BioRisk> riskMap = bioRiskService.loadBioRiskMap(codes);
+        List<String> unknown = codes.stream()
+                .filter(code -> riskMap.get(code) == null)
+                .map(BasicUtils::repr)
+                .toList();
+        if (!unknown.isEmpty()) {
+            addProblem("Unknown bio risks: " + unknown);
+        }
+        return riskMap;
+    }
+
```

Comment:
This is quite similar to the registration bio-risk validation. Can we reuse the same code to reduce redundancy?

Reply:
:+1: I've abstracted the code to BioRiskService.loadAndValidateBioRisks